src/recon_buildblock/RelativeDifferencePrior.cxx

```diff
@@ -0,0 +1,705 @@
+      
+      const int min_y = current_image_estimate[z].get_min_index(); 
+      const int max_y = current_image_estimate[z].get_max_index(); 
+      
+      for (int y=min_y;y<= max_y;y++) 
+        { 
+          const int min_dy = max(weights[0].get_min_index(), min_y-y); 
+          const int max_dy = min(weights[0].get_max_index(), max_y-y);             
+          
+          const int min_x = current_image_estimate[z][y].get_min_index();
+          const int max_x = current_image_estimate[z][y].get_max_index();  
+          
+          for (int x=min_x;x<= max_x;x++)
+            {
+              const int min_dx = max(weights[0][0].get_min_index(), min_x-x);
+              const int max_dx = min(weights[0][0].get_max_index(), max_x-x);
+                
+                /* Relative Difference Prior Gradient given by Eq.6 of J. Nuyts, D. Bequ, P. Dupont, and L. Mortelmans,
+                   “A Concave Prior Penalizing Relative Differences for Maximum-a-Posteriori Reconstruction
+                   in Emission Tomography,” vol. 49, no. 1, pp. 56–60, 2002.
+
+                formula:
+                  weights[dz][dy][dx] *
+                  (((current_image_estimate[z][y][x] - current_image_estimate[z+dz][y+dy][x+dx]) *
+                   (this->gamma * abs(current_image_estimate[z][y][x] - current_image_estimate[z+dz][y+dy][x+dx])) +
+                   (current_image_estimate[z][y][x] + 3* current_image_estimate[z+dz][y+dy][x+dx]))/
+                  square((current_image_estimate[z][y][x] + current_image_estimate[z+dz][y+dy][x+dx]) +
+                    this->gamma * abs(current_image_estimate[z][y][x] - current_image_estimate[z+dz][y+dy][x+dx]))) *
+                   (*kappa_ptr)[z][y][x] * (*kappa_ptr)[z+dz][y+dy][x+dx];
+                */
+
+#if 1
+                elemT gradient = 0;
+                for (int dz=min_dz;dz<=max_dz;++dz)
+                  for (int dy=min_dy;dy<=max_dy;++dy)
+                    for (int dx=min_dx;dx<=max_dx;++dx)
+                      {
+
+                        elemT current;
+                            current = weights[dz][dy][dx] *
+                                    (((current_image_estimate[z][y][x] - current_image_estimate[z+dz][y+dy][x+dx]) *
+                                      (this->gamma * abs(current_image_estimate[z][y][x] - current_image_estimate[z+dz][y+dy][x+dx]) +
+                                       current_image_estimate[z][y][x] + 3 * current_image_estimate[z+dz][y+dy][x+dx]))/
+                                     (square((current_image_estimate[z][y][x] + current_image_estimate[z+dz][y+dy][x+dx]) +
+                                      this->gamma * abs(current_image_estimate[z][y][x] - current_image_estimate[z+dz][y+dy][x+dx])) +small));
+                        if (do_kappa)
+                          current *= (*kappa_ptr)[z][y][x] * (*kappa_ptr)[z+dz][y+dy][x+dx];
+
+                        gradient += current;
+                      }
+#else
```

Comment:
delete all this, as it'd be incorrect for the RDP

Reply:
Can I please verify that you mean everything in the `#else` statement?

Reply:
yes